buildimage.go

```diff
@@ -54,6 +62,14 @@ func extraBinMust(n string) string {
 	return p
 }
 func main() {
+	//if GO111MODULE=off or GO111MODULE=on, buildimage won't work
+	//only GO111MODULE="" or GO111MODULE=auto works
+	s := precheck("GO111MODULE")
+
+	if s != "" || s != "auto" {
+		//set GO111MODULE=auto if it's not already, or GO111MODULE="".
+		os.Setenv("GO111MODULE", "auto")
```

Comment:
you should not os.Setenv, rather, add the correct value to the Env in commands you run.